Dockerfile

```diff
@@ -46,19 +46,121 @@ RUN uv pip install torch==${TORCH_VERSION} --index-url https://download.pytorch.
+    && curl -sS https://bootstrap.pypa.io/get-pip.py | python${PYTHON_VERSION} \
+    && python3 --version && python3 -m pip --version \
+    && pip install uv \
+    && pip install -U packaging cmake ninja wheel setuptools pybind11 Cython \
+    && apt-get update && apt-get install -y git && apt-get install -y cmake  && apt-get install -y python3.10-venv \
+    && apt-get update && apt-get install -y openmpi-bin libopenmpi-dev
+
+RUN mkdir -p /app/install
+
+RUN git clone https://github.com/ROCm/hipBLAS-common.git \
+    && cd hipBLAS-common \
+    && git checkout ${HIPBLAS_COMMON_BRANCH} \
+    && mkdir build \
+    && cd build \
+    && cmake .. \
+    && make package \
+    && dpkg -i ./*.deb
+
+RUN git clone https://github.com/ROCm/hipBLASLt \
+    && cd hipBLASLt \
+    && git checkout ${HIPBLASLT_BRANCH} \
+    && ./install.sh -d --architecture ${PYTORCH_ROCM_ARCH} ${LEGACY_HIPBLASLT_OPTION} \
+    && cd build/release \
+    && make package -j$(nproc) \
+    && echo "Searching for build and .deb files:" \
+    && find . -type d -name "build" \
+    && find . -name "*.deb" 
+
+RUN cp /app/hipBLASLt/build/release/*.deb /app/hipBLAS-common/build/*.deb /app/install
+
+RUN git clone ${TRITON_REPO}
+RUN cd triton \
+    && git checkout ${TRITON_BRANCH} \
+    && cd python \
+    && python3 setup.py bdist_wheel --dist-dir=dist && \
+    pip install dist/*.whl
+RUN cp /app/triton/python/dist/*.whl /app/install
+
+# Set working directory
+WORKDIR /pytorch
+
+# Clone and build PyTorch
+RUN git clone ${PYTORCH_REPO} pytorch && \
+    cd pytorch && \
+    git checkout ${PYTORCH_BRANCH} && \
+    pip install -r requirements.txt && \
+    git submodule update --init --recursive && \
+    python3 tools/amd_build/build_amd.py && \
+    CMAKE_PREFIX_PATH=$(python3 -c 'import sys; print(sys.prefix)') python3 setup.py bdist_wheel --dist-dir=dist && \
+    pip install dist/*.whl
+
+# Clone and build torchvision
+RUN git clone ${PYTORCH_VISION_REPO} vision && \
+    cd vision && \
+    git checkout ${PYTORCH_VISION_BRANCH} && \
+    python3 setup.py bdist_wheel --dist-dir=dist && \
+    pip install dist/*.whl
+
+# Clone and build flash-attention
+RUN git clone ${FA_REPO} flash-attention && \
+    cd flash-attention && \
+    git checkout ${FA_BRANCH} && \
+    git submodule update --init && \
+    MAX_JOBS=${MAX_JOBS} GPU_ARCHS=${PYTORCH_ROCM_ARCH} python3 setup.py bdist_wheel --dist-dir=dist
+
+# Copy all wheel files to installation directory
+RUN cp /pytorch/pytorch/dist/*.whl /app/install && \
+    cp /pytorch/vision/dist/*.whl /app/install && \
+    cp /pytorch/flash-attention/dist/*.whl /app/install
 
-ENV PATH="/opt/conda/envs/py_3.10/bin:$PATH"
-ENV CONDA_PREFIX=/opt/conda/envs/py_3.10
-
-ARG MAX_JOBS=4
-
-RUN pip install uv
 
 ###############################################################################
 # VLLM BUILD STAGE
 FROM ${BASE_NAME} AS vllm
 
+# Use test command to check if the directory exists
```

Comment:
this is conditional to whether ROCM layer was built, and the hip libs were installed.